### PR TITLE
Add per-service release support

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Each service should be released separately and globally.

## Proposal

- [x] document the release process
- [x] implement release process in the CI

### CODE REVIEW

**Review by wilbrdt — COMMENTED** [review]
Looks fine to me!

File: docs/contribute.md [hunk 1]
```diff
@@ -108,3 +108,22 @@ Patience during this process is appreciated.
...
+| Back-end     | API (core)   | `v[0-9]+.[0-9]+.[0-9]+-api`          | `v2.0.1-api`        |
+|              | API (plugin) | `v[0-9]+.[0-9]+.[0-9]+-api-<plugin>` | `v0.6.22-api-video` |
+|              | APP          | `v[0-9]+.[0-9]+.[0-9]+-app`          | `v1.3.6-app`        |
+| Front-end    | core         | `v[0-9]+.[0-9]+.[0-9]+-ui`           | `v5.1.0-ui`         |
+|              | plugin       | `v[0-9]+.[0-9]+.[0-9]+-ui-<plugin>`  | `v3.22.5-ui-video`  |
+| Distribution | \*           | `v[0-9]+.[0-9]+.[0-9]`               | `v2.0.0`            |
+
+!!! Note
+
+    A Warren **distribution** is considered as a "meta" package corresponding to a consistant combination of services releases that are known to work well together.
```
Comment (lebaudantoine) [inline docs/contribute.md:129]:
```suggestion
    A Warren **distribution** is considered as a "meta" package corresponding to a consistent combination of services releases that are known to work well together.
```

File: docs/contribute.md [hunk 2]
```diff
@@ -108,3 +108,22 @@ Patience during this process is appreciated.
 
 Once your PR is approved, Warren maintainers will merge your changes into the main project.
 Congratulations, you've successfully contributed to Warren! 🎉
+
+## Releases
+
+The Warren project has multiple services maintained in a single Git repository (_aka_ a monorepo): each service has its own life cycle with its own releases.
+
+We use Git tags to trigger CI builds of the Warren's artifacts (PyPI/NPM pachages and Docker images). To make a new release, depending on the service your are releasing, you need to apply the following Git tag pattern conventions:
```
Comment (lebaudantoine) [inline docs/contribute.md:116]:
```suggestion
The Warren project has multiple services maintained in a single Git repository (_aka_ a monorepo): each service has its own life cycle with its own releases.

We use Git tags to trigger CI builds of the Warren's artifacts (PyPI/NPM packages and Docker images). To make a new release, depending on the service you are releasing, you need to apply the following Git tag pattern conventions:
```

**Review by lebaudantoine — COMMENTED** [review]
Small typosss, but thx for your work mindblowing 🤯

File: docs/contribute.md [hunk 3]
```diff
@@ -108,3 +108,34 @@ Patience during this process is appreciated.
...
+| Type         | Service      | Git tag pattern                      | Example             |
+| ------------ | ------------ | ------------------------------------ | ------------------- |
+| Back-end     | API (core)   | `v[0-9]+.[0-9]+.[0-9]+-api`          | `v2.0.1-api`        |
+|              | API (plugin) | `v[0-9]+.[0-9]+.[0-9]+-api-<plugin>` | `v0.6.22-api-video` |
+|              | APP          | `v[0-9]+.[0-9]+.[0-9]+-app`          | `v1.3.6-app`        |
+| Front-end    | core         | `v[0-9]+.[0-9]+.[0-9]+-ui`           | `v5.1.0-ui`         |
+|              | plugin       | `v[0-9]+.[0-9]+.[0-9]+-ui-<plugin>`  | `v3.22.5-ui-video`  |
+| Distribution | \*           | `v[0-9]+.[0-9]+.[0-9]+`               | `v2.0.0`            |
+
+!!! Note
```
Comment (quitterie-lcs) [inline docs/contribute.md:127]: Could you add a short title to the Note ?

**Review by quitterie-lcs — APPROVED** [review]

**Pull request merged**